services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -672,6 +687,28 @@ public void handleDeletedInstances(Set<Instance> deletedInstances) {
     }
   }
 
+  private Optional<String> discoverClusterFromStore(PubSubTopic pubSubTopic) {
+    if (pubSubTopic.getPubSubTopicType().equals(PubSubTopicType.ADMIN_TOPIC)) {
+      return Optional.of(AdminTopicUtils.getClusterNameFromTopicName(pubSubTopic.getName()));
+    }
+    String storeName = pubSubTopic.getStoreName();
+    if (VeniceSystemStoreUtils.isSystemStore(storeName)) {
+      if (storeName.equals(AvroProtocolDefinition.METADATA_SYSTEM_SCHEMA_STORE.getSystemStoreName())
+          || storeName.equals(AvroProtocolDefinition.PUSH_STATUS_SYSTEM_SCHEMA_STORE.getSystemStoreName())) {
+        return Optional.of(multiClusterConfigs.getSystemSchemaClusterName());
+      }
+    }
+
+    try {
+      String clusterName = discoverCluster(pubSubTopic.getStoreName()).getFirst();
+      LOGGER.debug("Discovered cluster {} from store for topic {}", clusterName, pubSubTopic);
```

Comment:
debug level logging is not printed by default

Reply:
As this path will be called very frequently, better to keep it as debug level.